text/0000-exception-safe-event-handlers.md

```diff
@@ -0,0 +1,164 @@
+- Feature Name: exception_safe_event_handlers
+- Start Date: 2020-01-31
+- RFC PR: 
+- Neon Issue: 
+
+# Summary
+[summary]: #summary
+
+[RFC 25](https://github.com/neon-bindings/rfcs/blob/master/text/0025-event-handler.md) introduced a powerful new API for transmitting a JavaScript event handler to other Rust threads so that they can asynchronously signal events back to the main JavaScript thread. This RFC proposes a modification to the API to make the event handler's callback protocol both **simpler to understand** and **safe for handling JavaScript exceptions** that can occur while preparing the call the callback.
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 25](https://github.com/neon-bindings/rfcs/blob/master/text/0025-event-handler.md) introduced a powerful new API for transmitting a JavaScript event handler to other Rust threads so that they can asynchronously signal events back to the main JavaScript thread.
+
+It offers a low-level API, which gives the Rust thread complete control in how it interacts with the callback:
+
+```rust
+// Main thread:
+let this = cx.undefined();
+let handler = EventHandler::new(cx, this, f);
+
+// ...
+
+// Another Rust thread:
+handler.schedule_with(move |cx, this, f| {
+    let args = vec![ /* ... */ ];
+    f.call(cx, this, args).unwrap();
```

Comment:
This usage doesn't appear to match the API. Should this changed to...

```rust
handler.schedule_with(move |cx, this, f| {
    let args = vec![ /* ... */ ];
    f.call(cx, this, args)?;

    Ok(())
});
```

Reply:
This is describing the API as it appears in RFC 25, but that's not very clear. Let me tighten this section up a bit.